Bluetooth: Controller: Fix BT_CTLR_ADV_DATA_LEN_MAX Kconfig dependency

cvinayak opened the pull request:
Fix BT_CTLR_ADV_DATA_LEN_MAX and other Kconfig dependencies,
and clean up the use of PDU_AC_LL_SIZE_MAX which is used to
allocate scratch packet to be atleast able to accommodate
the minimum 31 bytes AD data such that it is sufficient to
transmit and/or receive scan response PDUs when either
broadcaster or observer alone is supported.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

cvinayak committed:
Bluetooth: Controller: Fix BT_CTLR_ADV_DATA_LEN_MAX Kconfig dependency

Fix BT_CTLR_ADV_DATA_LEN_MAX and other Kconfig dependencies,
and clean up the use of PDU_AC_LL_SIZE_MAX which is used to
allocate scratch packet to be atleast able to accommodate
the minimum 31 bytes AD data such that it is sufficient to
transmit and/or receive scan response PDUs when either
broadcaster or observer alone is supported.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

Thalley commented on subsys/bluetooth/controller/ll_sw/pdu.h:
```diff
@@ -36,11 +36,19 @@
+#if defined(CONFIG_BT_BROADCASTER)
+#define BT_CTLR_ADV_DATA_LEN_MAX CONFIG_BT_CTLR_ADV_DATA_LEN_MAX
+#else /* !CONFIG_BT_BROADCASTER */
+#define BT_CTLR_ADV_DATA_LEN_MAX PDU_AC_LEG_DATA_SIZE_MAX
```
Why do we need to define `BT_CTLR_ADV_DATA_LEN_MAX` if `CONFIG_BT_BROADCASTER` is not enabled?

Thalley commented on subsys/bluetooth/controller/ll_sw/ull_adv.c:
```diff
@@ -1723,7 +1723,7 @@ uint8_t ull_adv_data_set(struct ll_adv_set *adv, uint8_t len,
 
 	/* Check invalid AD Data length */
-	if (len > PDU_AC_DATA_SIZE_MAX) {
+	if (len > PDU_AC_LEG_DATA_SIZE_MAX) {
```
`ull_adv_data_set` seems to also be used when setting extended advertising data; checking just the legacy data size max seems odd to do in that case.

Thalley replied:
After checking again, you are correct :)

wopu-ot reviewed (dismissed):
LGTM

cvinayak commented:
@Thalley @ppryga-nordic @wopu-ot rebased to resolve conflicts.

cvinayak committed:
Bluetooth: Controller: Add Kconfig for experimental AD Data Chaining

Add a Kconfig for Advertising Data Chaining in Extended and
Periodic Advertising, permitting AD data lengths of upto
1650 bytes. The implementation is experimental and needs
further testing.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

Thalley commented on subsys/bluetooth/controller/Kconfig.ll_sw_split:
```diff
@@ -189,6 +189,22 @@ config BT_CTLR_ADV_SYNC_SET
 	  Maximum supported periodic advertising sets.
 
+config BT_CTLR_ADV_DATA_CHAIN
+	bool "Advertising Data chaining"
```
We've typically added "[EXPERIMENTAL]" for experimental Kconfigs

```suggestion
	bool "Advertising Data chaining [EXPERIMENTAL]"
```

Thalley commented on tests/bluetooth/bsim_bt/bsim_test_adv_chain/prj.conf:
```diff
@@ -5,6 +5,11 @@ CONFIG_BT_EXT_ADV=y
 CONFIG_BT_EXT_ADV_MAX_ADV_SET=2
 CONFIG_BT_DEVICE_NAME="Broadcaster Multiple"
 
+# Enable the experimental Advertising Data chaining in Zephyr Bluetooth LE
```
```suggestion
# Enable the Advertising Data chaining in Zephyr Bluetooth LE
```

Suggest to remove experimental so that we don't need to remember to modify this comment once "[EXPERIMENTAL]" is removed.

cvinayak replied:
done
